export/wiki/items/poc_stat_gathering.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any, Dict
+
+from ark.types import PrimalItem
+from ue.asset import ExportTableItem
+from ue.hierarchy import find_parent_classes
+
+DEFAULTS = {
+    # DevKit Verified
+    'bUsed': False,
+    'InitialValueConstant': 0,
+    'StateModifierScale': 1,
+    'DefaultModifierValue': 0,
+    'AbsoluteMaxValue': 0,
+}
+
+STAT_GENERIC_QUALITY = 0
+STAT_ARMOR = 1
+STAT_DURABILITY = 2
+STAT_DAMAGE_PERCENT = 3
+STAT_CLIP_AMMO = 4
+STAT_HYPO_INSULATION = 5
+STAT_WEIGHT = 6
+STAT_HYPER_INSULATION = 7
+
+
+def _gather_props_from_export(export: ExportTableItem, index: int, output: Dict[str, Any]):
+    props = export.properties.get_property('ItemStatInfos', index, fallback=None)
+    if not props:
+        return
+
+    for key, value in props.as_dict().items():
+        output[key] = value
+
+
+def gather_item_stat(item: PrimalItem, index: int) -> Dict[str, Any]:
+    leaf_export = item.get_source()
+    loader = leaf_export.asset.loader
+    output = dict(**DEFAULTS)
+
+    # Make sure this is always the default export, and not e.g. default class.
+    leaf_export = leaf_export.asset.default_export
+
+    for kls in reversed(list(find_parent_classes(leaf_export))):
+
+        if not kls.startswith('/Game'):
+            continue
+
+        asset = loader[kls]
+        super_export = asset.default_export
```

Comment:
By always using the default export you are doing each default export twice (once for the class, once for its defaults), and assuming there's never any useful properties on the class itself (which seems like a safe assumption currently).
Might be cleaner just to use `loader.load_class(kls)` and use that directly.

Reply:
`find_parent_classes` returns here blueprint paths for the classes, and not the CDOs - I do not believe `load_class` will be suitable for this (as it will load only the UClass export in this scenario).

It did, however, collect from the input CDO twice though, so that bit should be fixed now.